core/src/com/unciv/models/ruleset/tech/Technology.kt

```diff
@@ -81,47 +90,161 @@ class Technology {
         return lineList.joinToString("\n") { it.tr() }
     }
 
-    fun getEnabledBuildings(civInfo: CivilizationInfo): List<Building> {
-        var enabledBuildings = civInfo.gameInfo.ruleSet.buildings.values.filter {
-            it.requiredTech == name &&
-                    (it.uniqueTo == null || it.uniqueTo == civInfo.civName)
+    /**
+     * Returns a Sequence of [Building]s enabled by this Technology, filtered for [civInfo]'s uniques,
+     * nuclear weapons and religion settings, and without those expressly hidden from Civilopedia.
+     */
+    // Used for Civilopedia, Alert and Picker, so if any of these decide to ignore the "Will not be displayed in Civilopedia" unique this needs refactoring
+    fun getEnabledBuildings(civInfo: CivilizationInfo) = getFilteredBuildings(civInfo)
+        { it.requiredTech == name }
+
+    /**
+     * Returns a Sequence of [Building]s obsoleted by this Technology, filtered for [civInfo]'s uniques,
+     * nuclear weapons and religion settings, and without those expressly hidden from Civilopedia.
+     */
+    // Used for Civilopedia, Alert and Picker, so if any of these decide to ignore the "Will not be displayed in Civilopedia" unique this needs refactoring
+    fun getObsoletedBuildings(civInfo: CivilizationInfo) = getFilteredBuildings(civInfo)
+        { it.uniqueObjects.any { unique -> unique.placeholderText == "Obsolete with []" && unique.params[0] == name } }
+    
+    // Helper: common filtering for both getEnabledBuildings and getObsoletedBuildings, difference via predicate parameter
+    private fun getFilteredBuildings(civInfo: CivilizationInfo, predicate: (Building)->Boolean): Sequence<Building> {
+        val nuclearWeaponsEnabled = civInfo.gameInfo.gameParameters.nuclearWeaponsEnabled
+        val religionEnabled = civInfo.gameInfo.hasReligionEnabled()
+
+        return civInfo.gameInfo.ruleSet.buildings.values.asSequence()
+            .filter {
+                predicate(it)   // expected to be the most selective, thus tested first
```

Comment:
I wonder, this looks like it applies to buildings in the Civilopedia on general.
Maybe building should have a "should be displayed in Civilopedia" function that accepts a civinfo?